src/components/Table/utils/columns.tsx

```diff
@@ -146,3 +148,10 @@ export function calcColumnSizes(
 
   return sizes;
 }
+
+/** Assign column ids if missing */
+export function assignDefaultColumnIds<T extends Kinded>(columns: GridColumn<T>[]): GridColumnWithId<T>[] {
+  return columns.map((c, idx) => (c.id ? (c as GridColumnWithId<T>) : { ...c, id: c.id ?? generateColumnId(idx) }));
```

Comment:
Suggestion: Could do

```
  return columns.map((c, idx) => ({ id: generateColumnId(idx), ...c }));
```

To avoid the `as`. Granted it always makes a copy but :shrug: this is in a useMemo so should be really cheap.

Reply:
I had that before, but the spread breaks the proxy for the selectColumn and collapseColumn. I can put an in source comment for why we aren't always spreading.

Reply:
Thanks for the in-source column; what aspect of proxies was breaking for this?

I didn't think we made the columns observable, so initially am not following where a proxy would come in; not surprised that there is, just curious where/what it was.

Reply:
It was the `newMethodMissingProxy`. Basically removing wasn't able to generate the new methods any longer.